### DISCUSSION TITLE
Document links

### DISCUSSION THREAD

**Discussion by jool-kba** [hajkmap/Hajk#1211] [Q&A]
Hi!

I've read [here](https://github.com/hajkmap/Hajk/wiki/Lagerhantering) (5) that links to documents with map links can be rendered in layer info. Is this feature missing in hajk3 or is it a bug that prevents the rendering? I'm currently converting old documents versions to documenthandler 2.0 while trying to minimise changes between the versions. Would be great to get the document link rendering feature back in hajk3. In addition I would like to see support for adding tag@@documenthandler links to layer info.

**Comment by jacobwod** [comment]
I can't see it's missing, if you're referring to this feature:
![image](https://user-images.githubusercontent.com/110222/195604364-1ec0dbfd-61d3-47ab-a0da-0886c4252b54.png)


We use it frequently on e.g. https://framtidsplan2050antagande.halmstad.se.

**Reply by jool-kba** [reply]
I'm referring to the layer switcher. Have a look at [karta.kungsbacka.se/oversiktsplan](https://karta.kungsbacka.se/oversiktsplan).
![image](https://user-images.githubusercontent.com/100673923/195623771-d3c787a2-e693-4a9c-b80b-c3bb0a8bd43d.png)

**Reply by jacobwod** [reply]
OK now I see. The old version you refer to is using the deprecated `Informative` plugin. The functionality of having links from LayerSwitcher back to the document window was not considered something to port when the new `DocumentHandler` was developed. Instead the map-to-text connection is made via infoclick window (see my first post).

I _highly_ recommend migrating your `Informative` solution to `DocumentHandler` where possible, while keeping a separate legacy instance of Hajk running `Informative` (and nothing else) on those solutions that can not be migrated (due to formal reasons such as policies etc).

**Comment by jool-kba** [comment]
Ok. I've been trying to avoid the info-click solution when migrating data to reduce work.

**Reply by jacobwod** [reply]
I see. We must though keep in mind that _Informative_ and _DocumentHandler_ are two different plugins. Simple one-to-one upgrade is (as you've noticed) not possible. The correct way of solving this is - at least for me - keeping the legacy _Informative_ solution on a separate legacy Hajk instance, while the new solutions should be created with DocumentHandler.
